src/streamlink_cli/argparser.py

```diff
@@ -635,6 +635,21 @@ def build_parser():
     )
 
     transport = parser.add_argument_group("Stream transport options")
+    transport.add_argument(
+        "--dash-segment-stream-data",
```

Comment:
I can look into properly supporting this if that's what would be needed in the long run? I don't have much else on my plate at the moment, so I can put some time into contributing to streamlink.

Reply:
@back-to when you say using `http.get(..., stream=True)` will cause playback issues, does that mean in general for streamlink, or should it be safe to use for DASH segmentBase streams since there is only one segment per stream?

When I test some changes along the lines of

1. Fetch segmentBase index/initialization range as one streamlink Segment
2. Fetch the rest of the of the segmentBase data as another streamlink Segment with `http.get(stream=True)` to force immediate playback

I don't have any issues with streamlink closing nicely when I close my media player (mpv).

Reply:
> when you say using http.get(..., stream=True) will cause playback issues,

last time I used it with segmentBase, it did not work correctly and would just stuck until ctrl+c

> I don't have any issues with streamlink closing nicely when I close my media player (mpv).

well share the code, so it can be tested 
https://gist.github.com/ or update this PR

---

> Fetch the rest of the of the segmentBase data as another streamlink Segment with http.get(stream=True) to force immediate playback

generating multiple with the size of the initialization, same as it is done in the browser might be more stable